coralproject/talk
# rate limit testSMTP mutation

## What does this PR do?
This PR limits the number of times a user can send a test email from the admin config to once every 10 seconds.

## What changes to the GraphQL/Database Schema does this PR introduce?
The `testSMTP` mutation is decorated with the `@rate()` directive

## Does this PR introduce any new environment variables or feature flags? 
No.

## If any indexes were added, were they added to `INDEXES.md`?
n/a

## How do I test this PR?
1. Set `DISABLE_RATE_LIMITERS=false`
2. Start up coral.
3. log in as an admin, andNavigate to admin > configure > email
4. Send a test email, and then try again after < 10 seconds. 

You should see a message indicating you need to wait to try again. If you try again after >= 10 seconds (you will need to refresh the page or navigate away to reenable the "Send Test Email" button) it will successfully send an email.
